fixes Book1/gauss_plot_2d

Aadesh-1404 opened the pull request:
# This PR is not ready for merge 

## Description

<!-- Please refer to https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md before opening this PR -->

Hello Dr @murphyk,
I am Aadesh Desai, working with Prof @nipunbatra and the lab colleagues @patel-zeel and @karm-patel. I have updated gauss_plot_2d.ipynb in Book 1 based on the contribution guidelines and feedback from fellow colleagues. Kindly find time to review the notebook

Added gauss_plot_2d.ipynb for  [gauss_plot_2d.ipynb](https://github.com/Aadesh-1404/pyprobml/blob/master/notebooks/book1/03/gauss_plot_2d.ipynb)

### Figure Number

Book1 -  Fig 3.5

### Figures

[gist](https://gist.github.com/Aadesh-1404/f354f55888c6fc86388c9c3b1ebeb28f)

### Issue 
#739 

<!-- If the issue is from this repo, include directly with issue number -->
<!-- For example: 

#12

-->

<!-- If the issue is from another repo, you can include it using the regular linking mechanism -->
<!-- For example:

[Issue title](Issue link) 

-->

## Steps

<!-- If the task is complicated, you can specify each step using checkboxes so that everyone can understand what you have done and what you are going to complete. -->
<!-- For example:
-->

- [x]  Changed the name of the variables/functions to PEP 8 style
- [x] Converted code to JAX
- [x]  Added code to generate figure as given in the book and Latexified them.

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

## Potential problems/Important remarks

 * Labeled the axes and updated the code to make the 2d probability distribution surface plot more clear.

<!-- If you have any important remarks for the reviewers to take a closer look, you can write them in detail here.  -->

murphyk commented:
This looks good. However the following code is not very JAX
```
#Multivariate gaussian PDF
class gaussian_pdf(object):


    def multi_normal(self, x, y, G):
        return multivariate_normal.pdf(jnp.array([x, y]), mean=mean, cov=covariance[G])

#Vectorizing 
multivariate_pdfs = gaussian_pdf()
gauss_pdf = jnp.vectorize(multivariate_pdfs.multi_normal, excluded=[2])
```
There is no need to make a class object. Typically when evaluating a distribution on a batch of points
you just vmap the log density function across the batch.

Aadesh-1404 commented:
<!-- Please refer to https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md before opening this PR -->

Dr @murphyk, thank you for the comments, I have updated gauss_plot_2d.ipynb in Book 1 based on the feedback received. Also, as Fig 3.6 has the same notebook, I have also uploaded the gist of it. Kindly find time to review the notebook.

The notebook contains the figures: Book1 -  Fig 3.5 and Fig 3.6

The gist and issue for both figures can be found below

### Figures

[gist](https://gist.github.com/Aadesh-1404/f354f55888c6fc86388c9c3b1ebeb28f)

### Issue 
#739 

<!-- If the issue is from this repo, include directly with issue number -->
<!-- For example: 

#12

-->

<!-- If the issue is from another repo, you can include it using the regular linking mechanism -->
<!-- For example:

[Issue title](Issue link) 

-->


## Potential problems/Important remarks

 * I have removed the class and also used vmap as suggested
 * Labeled the axes and updated the code to make the 2d probability distribution surface plot more clear. 
 * For Figure 3.6, added a colorbar for a better understanding of the contours

<!-- If you have any important remarks for the reviewers to take a closer look, you can write them in detail here.  -->
